sdk/core/Azure.Core/src/Operation.cs

```diff
@@ -25,7 +26,7 @@ public abstract class Operation
         /// <param name="rehydrationToken">The rehydration token.</param>
         /// <param name="options">The client options.</param>
         /// <returns>The long-running operation.</returns>
-        public static Operation<T> Rehydrate<T>(HttpPipeline pipeline, RehydrationToken rehydrationToken, ClientOptions? options = null) where T : IPersistableModel<T>
+        public static Operation<T> Rehydrate<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.NonPublicConstructors)] T>(HttpPipeline pipeline, RehydrationToken rehydrationToken, ClientOptions? options = null) where T : IPersistableModel<T>
```

Comment:
It would be good to make sure that these attributes appear in the Azure.Core APIView and that architects are aware of them when they are asked to approve the public APIs for GA.

Do we plan to remove these attributes when we make the types public, or do these modify the type parameter and so will always be there because we use MRW in all cases?  If the latter, that seems reasonable.  If the former, we should assess whether we plan to remove these later and if that would be a breaking change.

Reply:
I don't know what MRW stands for, but in general the attributes should stay when the types are made public to support AOT compat

Reply:
The attributes are always needed since we are using ModelReaderWriter here.

Reply:
> It would be good to make sure that these attributes appear in the Azure.Core APIView and that architects are aware of them when they are asked to approve the public APIs for GA.

The attributes do not appear in the api files, so won't show in APIView.
Existing MRW signature
https://github.com/Azure/azure-sdk-for-net/blob/410c18b8b5b853967e96d3618100e662cbd3d537/sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriter.cs#L94
Existing api file
https://github.com/Azure/azure-sdk-for-net/blob/410c18b8b5b853967e96d3618100e662cbd3d537/sdk/core/System.ClientModel/api/System.ClientModel.netstandard2.0.cs#L158